### ISSUE TITLE
Unable to reach main content when using keyboard navigation

### ISSUE THREAD

**Issue by ehoops-cz** [readthedocs/readthedocs.org#10835]
## Details

* Read the Docs project URL: https://docs.readthedocs.io/en/stable/
* Build URL (if applicable): n/a
* Read the Docs username (if applicable): n/a

## Expected Result

When navigating with the keyboard, I should be able to reach the "Tutorials" navigation header. When the search modal opens, the X button should be reachable with the `Tab` key.  I should be able to close the modal with the X button or with the `Escape` key.  In either case, focus should allow me to continue tabbing down the page after closing the modal.

## Actual Result

Pressing tab twice opens the search modal.  The X button within the modal is not reachable with the `Tab` key.  However, the modal does close using the `Escape` key. When the modal closes, focus goes back to the beginning of the page - this makes it impossible to get past the search box using the keyboard.

![stuck-in-search](https://github.com/readthedocs/readthedocs.org/assets/109251328/8e23d62f-f6ed-46f2-844b-2a9d2d6c69e3)

**Comment by ehoops-cz** [comment]
I think I can put up a fix for this if you are open to it.

**Comment by stsewd** [comment]
@ehoops-cz hi, thanks for reporting this and offering to work on a fix. This is a bug from our search extension (https://github.com/readthedocs/readthedocs-sphinx-search). But we are considering archiving that repository, and focus more on a generic search integration at https://github.com/readthedocs/addons#search-as-you-type. The new integration has probably some of the same bugs described in this issue, but we are focussing on fixing them there instead of the sphinx extension.

**Comment by ehoops-cz** [comment]
Thanks @stsewd.  Not making fixes in the sphinx extension makes sense.  I'd like to do some more accessibility auditing.  If there's portion of the site that you would like eyes on, let me know.
